File: programmers/js/level00/120902.js
// 문자열 계산하기
function solution(my_string) {
  let operator = 1;

  return my_string
    .split(' ')
    .reduce((acc, cur) => {
      if (!isNaN(cur)) {
        return acc + (+cur * operator);
      } else {
        operator = cur === '+' ? 1 : -1;
        return acc;
      }
    }, 0);
}

File: programmers/java/level00/120902.java
// 문자열 계산하기

class Solution {
    public int solution(String my_string) {
        int result = 0;
        int operator = 1;
        
        for (String val : my_string.split(" ")) {
            if (val.equals("+")) {
                operator = 1;
            } else if (val.equals("-")) {
                operator = -1;
            } else {
                result += Integer.parseInt(val) * operator;
            }
        }
        
        return result;
    }
}